components/dash-core-components/tests/integration/graph/test_graph_varia.py

```diff
@@ -193,6 +194,37 @@ def num_traces():
     assert dash_dcc.get_logs() == []
 
 
+def get_graph_points(browser, graph_id, expected_len):
+    until(
+        lambda: len(browser.find_elements(f"{graph_id} .points path")) == expected_len,
+        5,
+    )
+
+    traces = browser.find_elements(f"{graph_id} .points")
+
+    for trace in traces:
+        elements = trace.find_elements(By.CSS_SELECTOR, "path")
+        for element in elements:
+            ActionChains(browser.driver).move_to_element(element).perform()
+            hover = browser.find_element(f"{graph_id} .hoverlayer")
+            content = hover.text
```

Comment:
That'll work, though perhaps an easier solution (and more robust against changes in plotly.js implementation details) would be to write a little JS to look at `graphDiv._fullData[i].(x|y)`, where `graphDiv` is `document.querySelector("#{graph_id} .js-plotly-plot")`

Reply:
As discussed elsewhere, we can leave this for now, I may come back to it later myself.